libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationOwner.kt

```diff
@@ -0,0 +1,75 @@
+package com.mapbox.navigation.core.lifecycle
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+import com.mapbox.navigation.utils.internal.logI
+import java.util.concurrent.CopyOnWriteArraySet
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class MapboxNavigationOwner {
+    internal lateinit var navigationOptions: NavigationOptions
+        private set
+
+    private val services = CopyOnWriteArraySet<MapboxNavigationObserver>()
+    private var mapboxNavigation: MapboxNavigation? = null
+    private var attached = false
+
+    internal val carAppLifecycleObserver = object : DefaultLifecycleObserver {
+
+        override fun onStart(owner: LifecycleOwner) {
+            logI(TAG, Message("onStart"))
+            check(!MapboxNavigationProvider.isCreated()) {
+                "MapboxNavigation should only be created by the MapboxNavigationOwner"
+            }
+            check(this@MapboxNavigationOwner::navigationOptions.isInitialized) {
+                "You must set the NavigationOptions with MapboxNavigationApp.setup"
+            }
```

Comment:
It was necessary earlier on, we had a strict "setup first" requirement. But have been iterating since then, and this check will never actually happen. I'm replacing it with a test to demonstrate that it works.

```
    @Test
    fun `verify setup can be called after LifecycleOwner is started`() {
        val firstObserver = mockk<MapboxNavigationObserver>(relaxUnitFun = true)
        val secondObserver = mockk<MapboxNavigationObserver>(relaxUnitFun = true)
        mapboxNavigationApp.registerObserver(firstObserver)
        mapboxNavigationApp.registerObserver(secondObserver)

        val testLifecycleOwner = CarAppLifecycleOwnerTest.TestLifecycleOwner()
        mapboxNavigationApp.attach(testLifecycleOwner)

        testLifecycleOwner.lifecycleRegistry.currentState = Lifecycle.State.RESUMED
        mapboxNavigationApp.setup(navigationOptions)

        verify(exactly = 1) { firstObserver.onAttached(any()) }
        verify(exactly = 1) { secondObserver.onAttached(any()) }
        verify(exactly = 0) { firstObserver.onDetached(any()) }
        verify(exactly = 0) { secondObserver.onDetached(any()) }
    }
```